cpp/include/cudf/detail/gather.cuh

```diff
@@ -155,7 +158,14 @@ struct column_gatherer
                                      bool ignore_out_of_bounds,
                                      rmm::mr::device_memory_resource *mr,
                                      cudaStream_t stream) {
-    CUDF_FAIL("Column type must be numeric");
+
+    if (source_column.type().id() == STRING) {
```

Comment:
This should just be an explicit specialization of `operator()` for `string_view`.

Reply:
Will there be any other type of columns which will not be fixed_width type column apart from string_view

Reply:
Eventually yes. Lists, structures, etc.

Reply:
> This should just be an explicit specialization of `operator()` for `string_view`.

Is it possible to have partial specialization as we still need `MapIterator`

Reply:
No, you can't have a partial specialization, but you can use `enable_if` to only enable it for `string_view`.

Reply:
I have used specialization by creating one more structure, have a look, if you feel it is little too complicated, will move to enable_if